### ISSUE TITLE
Bug in floor function

### ISSUE THREAD

**Issue by hwmland** [nodemcu/nodemcu-firmware#1453]
### Expected behavior
floor (n) of negative integer should be n 

### Actual behavior
floor (n) of negative integer returns (n - 1)

### Test code
```Lua
print (math.floor (-1)) --returns -2 instead of -1
print (-2 % 2) --returns 2 instead of 0 because of wrong floor implementation
```
### NodeMCU version
Actual master (commit 98d3b46e2c11068f361154a564ce75a048c9dadc) (float firmware)
Problem is in file https://github.com/nodemcu/nodemcu-firmware/blob/master/app/libc/c_math.c#L7

### Recomended solution (not tested!)
```C
double floor(double x)
{
    return (double) (x < 0.f ? ((int)x == x ? x :  (((int) x) - 1)) : ((int) x));
}
```

**Comment by marcelstoer** [comment]
Thanks for the proposed fix, makes sense: #1454 .

**Comment by jmattsson** [comment]
Fix merged into dev.
